--- gcc-linaro-snapshot-5.3-2016.03/gcc/cp/except.c.EVILaarch64~	2016-03-22 09:47:21.301481356 +0100
+++ gcc-linaro-snapshot-5.3-2016.03/gcc/cp/except.c	2016-03-22 09:48:05.271185252 +0100
@@ -1040,8 +1040,12 @@ nothrow_libfn_p (const_tree fn)
      unless the system headers are playing rename tricks, and if
      they are, we don't want to be confused by them.  */
   id = DECL_NAME (fn);
+#ifdef __aarch64__
+  return 0;
+#else
   return !!libc_name::libc_name_p (IDENTIFIER_POINTER (id),
 				   IDENTIFIER_LENGTH (id));
+#endif
 }
 
 /* Returns nonzero if an exception of type FROM will be caught by a
