--- gcc-linaro-4.7-2012.06/config/override.m4.bero	2012-07-01 12:11:24.152641110 +0200
+++ gcc-linaro-4.7-2012.06/config/override.m4	2012-07-01 12:11:42.214641199 +0200
@@ -29,7 +29,7 @@ m4_copy_force([_AC_PREREQ], [AC_PREREQ])
 
 dnl Ensure exactly this Autoconf version is used
 m4_ifndef([_GCC_AUTOCONF_VERSION],
-  [m4_define([_GCC_AUTOCONF_VERSION], [2.64])])
+  [m4_define([_GCC_AUTOCONF_VERSION], [2.69])])
 
 dnl Test for the exact version when AC_INIT is expanded.
 dnl This allows to update the tree in steps (for testing)
