--- gcc-6.2.0/gcc/ada/terminals.c.musl1~	2016-08-23 13:29:04.419054200 +0200
+++ gcc-6.2.0/gcc/ada/terminals.c	2016-08-23 15:08:12.301062257 +0200
@@ -1071,7 +1071,7 @@ __gnat_setup_winsize (void *desc, int ro
 /* On some system termio is either absent or including it will disable termios
    (HP-UX) */
 #if !defined (__hpux__) && !defined (BSD) && !defined (__APPLE__) \
-  && !defined (__rtems__)
+  && !defined (__rtems__) && !defined(__linux__)
 #   include <termio.h>
 #endif
 
--- gcc-6.2.0/libstdc++-v3/config/os/gnu-linux/os_defines.h.musl1~	2016-01-22 13:47:26.000000000 +0100
+++ gcc-6.2.0/libstdc++-v3/config/os/gnu-linux/os_defines.h	2016-08-23 13:29:04.419054200 +0200
@@ -41,7 +41,7 @@
 // Provide a declaration for the possibly deprecated gets function, as
 // glibc 2.15 and later does not declare gets for ISO C11 when
 // __GNU_SOURCE is defined.
-#if __GLIBC_PREREQ(2,15) && defined(_GNU_SOURCE)
+#if 1 // && defined(_GNU_SOURCE)
 # undef _GLIBCXX_HAVE_GETS
 #endif
 
