--- gcc-linaro-snapshot-6.3-2017.05/gcc/tsystem.h.omv~	2017-06-13 14:03:57.776266038 +0200
+++ gcc-linaro-snapshot-6.3-2017.05/gcc/tsystem.h	2017-06-13 14:04:14.288779540 +0200
@@ -23,6 +23,8 @@ a copy of the GCC Runtime Library Except
 see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 <http://www.gnu.org/licenses/>.  */
 
+#define __DEFINED_wchar_t 1
+#define __DEFINED_size_t 1
 #ifndef GCC_TSYSTEM_H
 #define GCC_TSYSTEM_H
 
