diff -up gcc-9.0.1-RC-20190430/config-ml.in.multi-do-libdir~ gcc-9.0.1-RC-20190430/config-ml.in
--- gcc-9.0.1-RC-20190430/config-ml.in.multi-do-libdir~	2019-05-01 21:10:34.146610869 +0200
+++ gcc-9.0.1-RC-20190430/config-ml.in	2019-05-01 21:11:31.931283693 +0200
@@ -511,6 +511,7 @@ multi-do:
 				ADAFLAGS="$(ADAFLAGS) $${flags}" \
 				prefix="$(prefix)" \
 				exec_prefix="$(exec_prefix)" \
+				libdir="$(libdir)" \
 				GOCFLAGS="$(GOCFLAGS) $${flags}" \
 				GDCFLAGS="$(GDCFLAGS) $${flags}" \
 				CXXFLAGS="$(CXXFLAGS) $${flags}" \
