diff -up gcc-12-20220320/libstdc++-v3/include/bits/locale_facets_nonio.tcc.omv~ gcc-12-20220320/libstdc++-v3/include/bits/locale_facets_nonio.tcc
--- gcc-12-20220320/libstdc++-v3/include/bits/locale_facets_nonio.tcc.omv~	2022-03-21 21:10:54.597212096 +0100
+++ gcc-12-20220320/libstdc++-v3/include/bits/locale_facets_nonio.tcc	2022-03-21 21:11:04.503031807 +0100
@@ -1465,7 +1465,7 @@ _GLIBCXX_END_NAMESPACE_LDBL_OR_CXX11
       ctype<_CharT> const& __ctype = use_facet<ctype<_CharT> >(__loc);
       __err = ios_base::goodbit;
       bool __use_state = false;
-#if __GNUC__ >= 5
+#if __GNUC__ >= 5 && !defined(__clang__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wpmf-conversions"
       // Nasty hack.  The C++ standard mandates that get invokes the do_get
