--- gcc-linaro-4.7-2012.06/gcc/configure.ac.bero	2012-07-01 22:35:17.698825392 +0200
+++ gcc-linaro-4.7-2012.06/gcc/configure.ac	2012-07-01 22:35:35.029825479 +0200
@@ -3369,7 +3369,7 @@ fi
 
 AC_MSG_CHECKING(linker plugin support)
 gcc_cv_lto_plugin=0
-if test -f liblto_plugin.la; then
+dnl if test -f liblto_plugin.la; then <--- This seems wrong...
   save_ld_ver="$ld_ver"
   save_ld_vers_major="$ld_vers_major"
   save_ld_vers_minor="$ld_vers_minor"
@@ -3417,7 +3417,7 @@ changequote([,])dnl
   ld_vers_major="$save_ld_vers_major"
   ld_vers_minor="$save_ld_vers_minor"
   ld_is_gold="$save_ld_is_gold"
-fi
+dnl fi
 AC_DEFINE_UNQUOTED(HAVE_LTO_PLUGIN, $gcc_cv_lto_plugin,
   [Define to the level of your linker's plugin support.])
 AC_MSG_RESULT($gcc_cv_lto_plugin)
