--- gcc-linaro-snapshot-6.2-2016.09/libjava/Makefile.am.omv~	2016-09-09 12:04:45.672564008 +0200
+++ gcc-linaro-snapshot-6.2-2016.09/libjava/Makefile.am	2016-09-09 12:05:12.593124890 +0200
@@ -501,7 +501,7 @@ libgcj_la_LIBADD = \
 	java/lang/Class.lo \
 	java/process-$(PLATFORM).lo \
 	$(libgcj_la_LIBADD_SUBOBJECTS) \
-	$(LIBFFI) $(ZLIBS) $(GCLIBS)
+	$(LIBFFI) $(ZLIBS) $(GCLIBS) -lstdc++
 libgcj_la_DEPENDENCIES = libgcj-$(gcc_version).jar \
 	java/lang/Object.lo \
 	java/lang/Class.lo \
--- gcc-linaro-snapshot-6.2-2016.09/libjava/Makefile.in.omv~	2016-09-09 12:05:42.276640680 +0200
+++ gcc-linaro-snapshot-6.2-2016.09/libjava/Makefile.in	2016-09-09 12:05:50.864500587 +0200
@@ -1116,7 +1116,7 @@ libgcj_la_LIBADD = \
 	java/lang/Class.lo \
 	java/process-$(PLATFORM).lo \
 	$(libgcj_la_LIBADD_SUBOBJECTS) \
-	$(LIBFFI) $(ZLIBS) $(GCLIBS)
+	$(LIBFFI) $(ZLIBS) $(GCLIBS) -lstdc++
 
 libgcj_la_DEPENDENCIES = libgcj-$(gcc_version).jar \
 	java/lang/Object.lo \
