--- gcc-linaro-snapshot-6.3-2017.06/Makefile.in.compare~	2017-07-07 20:39:30.682990437 +0200
+++ gcc-linaro-snapshot-6.3-2017.06/Makefile.in	2017-07-07 20:40:47.399216464 +0200
@@ -48801,7 +48801,7 @@ compare:
 	if [ -f .bad_compare ]; then \
 	  echo "Bootstrap comparison failure!"; \
 	  cat .bad_compare; \
-	  exit 1; \
+	  echo Please make sure everything works.; \
 	else \
 	  echo Comparison successful.; \
 	fi; \
@@ -49199,7 +49199,7 @@ compare3:
 	if [ -f .bad_compare ]; then \
 	  echo "Bootstrap comparison failure!"; \
 	  cat .bad_compare; \
-	  exit 1; \
+	  echo Please make sure everything works.; \
 	else \
 	  echo Comparison successful.; \
 	fi; \
--- gcc-linaro-snapshot-6.3-2017.06/Makefile.tpl.compare~	2017-07-07 20:39:30.684990391 +0200
+++ gcc-linaro-snapshot-6.3-2017.06/Makefile.tpl	2017-07-07 20:41:03.697839577 +0200
@@ -1646,7 +1646,7 @@ do-clean: clean-stage[+id+]
 	if [ -f .bad_compare ]; then \
 	  echo "Bootstrap comparison failure!"; \
 	  cat .bad_compare; \
-	  exit 1; \
+	  echo Please make sure everything works.; \
 	else \
 	  echo Comparison successful.; \
 	fi; \
