--- gcc-linaro-snapshot-5.1-2015.06/gcc/graphite.c.omv~	2015-07-17 11:23:41.877719891 +0200
+++ gcc-linaro-snapshot-5.1-2015.06/gcc/graphite.c	2015-07-17 11:23:51.352677633 +0200
@@ -39,6 +39,7 @@ along with GCC; see the file COPYING3.
 #include <isl/map.h>
 #include <isl/options.h>
 #include <isl/union_map.h>
+#include <isl/constraint.h>
 #endif
 
 #include "system.h"
--- gcc-linaro-snapshot-5.1-2015.06/gcc/graphite-dependences.c.omv~	2015-07-17 11:26:35.944942777 +0200
+++ gcc-linaro-snapshot-5.1-2015.06/gcc/graphite-dependences.c	2015-07-17 11:37:36.427343562 +0200
@@ -227,7 +227,7 @@ scop_get_transformed_schedule (scop_p sc
 /* Helper function used on each MAP of a isl_union_map.  Computes the
    maximal output dimension.  */
 
-static int
+static isl_stat
 max_number_of_out_dimensions (__isl_take isl_map *map, void *user)
 {
   int global_max = *((int *) user);
@@ -239,7 +239,7 @@ max_number_of_out_dimensions (__isl_take
 
   isl_map_free (map);
   isl_space_free (space);
-  return 0;
+  return isl_stat_ok;
 }
 
 /* Extends the output dimension of MAP to MAX dimensions.  */
@@ -263,12 +263,12 @@ struct extend_schedule_str {
 
 /* Helper function for extend_schedule.  */
 
-static int
+static isl_stat
 extend_schedule_1 (__isl_take isl_map *map, void *user)
 {
   struct extend_schedule_str *str = (struct extend_schedule_str *) user;
   str->umap = isl_union_map_add_map (str->umap, extend_map (map, str->max));
-  return 0;
+  return isl_stat_ok;
 }
 
 /* Return a relation that has uniform output dimensions.  */
--- gcc-linaro-snapshot-5.1-2015.06/gcc/graphite-interchange.c.omv~	2015-07-17 11:38:55.557038787 +0200
+++ gcc-linaro-snapshot-5.1-2015.06/gcc/graphite-interchange.c	2015-07-17 11:39:04.996002039 +0200
@@ -30,6 +30,7 @@ along with GCC; see the file COPYING3.
 #include <isl/union_map.h>
 #include <isl/ilp.h>
 #include <isl/val.h>
+#include <isl/constraint.h>
 
 /* Since ISL-0.13, the extern is in val_gmp.h.  */
 #if !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
--- gcc-linaro-snapshot-5.1-2015.06/gcc/graphite-isl-ast-to-gimple.c.omv~	2015-07-17 11:24:28.215513175 +0200
+++ gcc-linaro-snapshot-5.1-2015.06/gcc/graphite-isl-ast-to-gimple.c	2015-07-17 11:25:46.985161509 +0200
@@ -21,9 +21,11 @@ along with GCC; see the file COPYING3.
 #include "config.h"
 
 #ifdef HAVE_isl
+#include <isl/constraint.h>
 #include <isl/set.h>
 #include <isl/map.h>
 #include <isl/union_map.h>
+#include <isl/union_set.h>
 #include <isl/ast_build.h>
 
 /* Since ISL-0.13, the extern is in val_gmp.h.  */
--- gcc-linaro-snapshot-5.1-2015.06/gcc/graphite-optimize-isl.c.omv~	2015-07-17 11:39:50.010825727 +0200
+++ gcc-linaro-snapshot-5.1-2015.06/gcc/graphite-optimize-isl.c	2015-07-17 11:49:33.657431445 +0200
@@ -21,9 +21,11 @@ along with GCC; see the file COPYING3.
 #include "config.h"
 
 #ifdef HAVE_isl
+#include <isl/constraint.h>
 #include <isl/set.h>
 #include <isl/map.h>
 #include <isl/union_map.h>
+#include <isl/union_set.h>
 #include <isl/schedule.h>
 #include <isl/band.h>
 #include <isl/aff.h>
@@ -530,13 +532,13 @@ getScheduleMap (isl_schedule *Schedule,
   return ScheduleMap;
 }
 
-static int
+static isl_stat
 getSingleMap (__isl_take isl_map *map, void *user)
 {
   isl_map **singleMap = (isl_map **) user;
   *singleMap = map;
 
-  return 0;
+  return isl_stat_ok;
 }
 
 static void
@@ -608,7 +610,7 @@ optimize_isl (scop_p scop)
 
   isl_options_set_schedule_max_constant_term (scop->ctx, CONSTANT_BOUND);
   isl_options_set_schedule_maximize_band_depth (scop->ctx, 1);
-  isl_options_set_schedule_fuse (scop->ctx, ISL_SCHEDULE_FUSE_MIN);
+  isl_options_set_schedule_serialize_sccs (scop->ctx, 1);
   isl_options_set_on_error (scop->ctx, ISL_ON_ERROR_CONTINUE);
 
 #ifdef HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE
--- gcc-linaro-snapshot-5.1-2015.06/gcc/graphite-scop-detection.c.omv~	2015-07-17 11:54:36.176146016 +0200
+++ gcc-linaro-snapshot-5.1-2015.06/gcc/graphite-scop-detection.c	2015-07-17 11:54:44.521110331 +0200
@@ -25,6 +25,7 @@ along with GCC; see the file COPYING3.
 #include <isl/set.h>
 #include <isl/map.h>
 #include <isl/union_map.h>
+#include <isl/constraint.h>
 #endif
 
 #include "system.h"
