diff -up gcc-8.2.0/gcc/graphite.h.omv~ gcc-8.2.0/gcc/graphite.h
--- gcc-8.2.0/gcc/graphite.h.omv~	2018-09-11 16:22:54.761816619 +0200
+++ gcc-8.2.0/gcc/graphite.h	2018-09-11 16:23:13.985603165 +0200
@@ -37,6 +37,8 @@ along with GCC; see the file COPYING3.
 #include <isl/schedule.h>
 #include <isl/ast_build.h>
 #include <isl/schedule_node.h>
+#include <isl/id.h>
+#include <isl/space.h>
 
 typedef struct poly_dr *poly_dr_p;
 
