diff -up gcc-9.0.1-RC-20190430/gcc/ada/terminals.c.musl1~ gcc-9.0.1-RC-20190430/gcc/ada/terminals.c
diff -up gcc-9.0.1-RC-20190430/libstdc++-v3/config/os/gnu-linux/os_defines.h.musl1~ gcc-9.0.1-RC-20190430/libstdc++-v3/config/os/gnu-linux/os_defines.h
--- gcc-9.0.1-RC-20190430/libstdc++-v3/config/os/gnu-linux/os_defines.h.musl1~	2019-01-01 13:31:55.000000000 +0100
+++ gcc-9.0.1-RC-20190430/libstdc++-v3/config/os/gnu-linux/os_defines.h	2019-05-01 21:06:43.299063081 +0200
@@ -41,7 +41,7 @@
 // Provide a declaration for the possibly deprecated gets function, as
 // glibc 2.15 and later does not declare gets for ISO C11 when
 // __GNU_SOURCE is defined.
-#if __GLIBC_PREREQ(2,15) && defined(_GNU_SOURCE)
+#if 1 // && defined(_GNU_SOURCE)
 # undef _GLIBCXX_HAVE_GETS
 #endif
 
