--- gcc-linaro-7.1-2017.08-rc1/gcc/cp/except.c.EVILaarch64~	2017-08-09 20:12:38.299700509 +0200
+++ gcc-linaro-7.1-2017.08-rc1/gcc/cp/except.c	2017-08-09 20:14:51.656324364 +0200
@@ -896,6 +896,9 @@ nothrow_libfn_p (const_tree fn)
      unless the system headers are playing rename tricks, and if
      they are, we don't want to be confused by them.  */
   id = DECL_NAME (fn);
+#ifdef __aarch64__
+  return 0;
+#else
   const struct libc_name_struct *s
     = libc_name::libc_name_p (IDENTIFIER_POINTER (id), IDENTIFIER_LENGTH (id));
   if (s == NULL)
@@ -907,6 +910,7 @@ nothrow_libfn_p (const_tree fn)
     case 11: return !flag_iso || flag_isoc11;
     default: gcc_unreachable ();
     }
+#endif
 }
 
 /* Returns nonzero if an exception of type FROM will be caught by a
