diff -up gcc-9-20190706/gcc/collect2.c.omv~ gcc-9-20190706/gcc/collect2.c
--- gcc-9-20190706/gcc/collect2.c.omv~	2019-07-12 18:34:29.055363553 +0200
+++ gcc-9-20190706/gcc/collect2.c	2019-07-12 18:37:33.849374836 +0200
@@ -1087,6 +1087,10 @@ main (int argc, char **argv)
       lto_mode = LTO_MODE_LTO;
   }
 
+  /* LLD doesn't like gcc LTO at all -- so we fall back to BFD */
+  if (selected_linker == USE_DEFAULT_LD && lto_mode == LTO_MODE_NONE)
+    selected_linker = USE_BFD_LD;
+
   /* -fno-profile-arcs -fno-test-coverage -fno-branch-probabilities
      -fno-exceptions -w -fno-whole-program */
   num_c_args += 6;
