--- gcc-linaro-7.1-2017.08-rc1/config-ml.in.multi-do-libdir~	2017-08-09 20:20:17.937310686 +0200
+++ gcc-linaro-7.1-2017.08-rc1/config-ml.in	2017-08-09 20:25:04.640823156 +0200
@@ -511,6 +511,7 @@ multi-do:
 				ADAFLAGS="$(ADAFLAGS) $${flags}" \
 				prefix="$(prefix)" \
 				exec_prefix="$(exec_prefix)" \
+				libdir="$(libdir)" \
 				GOCFLAGS="$(GOCFLAGS) $${flags}" \
 				CXXFLAGS="$(CXXFLAGS) $${flags}" \
 				LIBCFLAGS="$(LIBCFLAGS) $${flags}" \
