--- gcc-linaro-4.9-2014.04/gcc/ada/gcc-interface/Makefile.in.execstack~	2014-04-16 14:44:16.595864247 +0200
+++ gcc-linaro-4.9-2014.04/gcc/ada/gcc-interface/Makefile.in	2014-04-16 14:46:31.577858721 +0200
@@ -210,7 +210,7 @@ RTSDIR = rts$(subst /,_,$(MULTISUBDIR))
 # Link flags used to build gnat tools.  By default we prefer to statically
 # link with libgcc to avoid a dependency on shared libgcc (which is tricky
 # to deal with as it may conflict with the libgcc provided by the system).
-GCC_LINK_FLAGS=-static-libstdc++ -static-libgcc
+GCC_LINK_FLAGS=-static-libstdc++ -static-libgcc -Wl,-z,execstack
 
 # End of variables for you to override.
 
