diff --git a/boehm-gc/include/private/gcconfig.h b/boehm-gc/include/private/gcconfig.h
index 475bb2c..6fe4a47 100644
--- a/boehm-gc/include/private/gcconfig.h
+++ b/boehm-gc/include/private/gcconfig.h
@@ -1846,6 +1846,7 @@
 # endif
 
 # ifdef AARCH64
+#   define USE_GENERIC_PUSH_REGS
 #   define CPP_WORDSZ 64
 #   define MACH_TYPE "AARCH64"
 #   define ALIGNMENT 8
@@ -1855,7 +1856,6 @@
 #   ifdef LINUX
 #     define OS_TYPE "LINUX"
 #     define LINUX_STACKBOTTOM
-#     define USE_GENERIC_PUSH_REGS
 #     define DYNAMIC_LOADING
       extern int __data_start[];
 #     define DATASTART ((ptr_t)__data_start)
diff --git a/configure b/configure
index 1e55feb..b2601cc 100755
--- a/configure
+++ b/configure
@@ -3309,6 +3309,8 @@ esac
 
 # Disable Java if libffi is not supported.
 case "${target}" in
+  aarch64*-*-*)
+    ;;
   alpha*-*-*)
     ;;
   arm*-*-*)
diff --git a/configure.ac b/configure.ac
index 637da1e..cf56677 100644
--- a/configure.ac
+++ b/configure.ac
@@ -611,6 +611,8 @@ esac
 
 # Disable Java if libffi is not supported.
 case "${target}" in
+  aarch64*-*-*)
+    ;;
   alpha*-*-*)
     ;;
   arm*-*-*)
diff --git a/libjava/classpath/native/fdlibm/ieeefp.h b/libjava/classpath/native/fdlibm/ieeefp.h
index c230bbb..7ef2ae7 100644
--- a/libjava/classpath/native/fdlibm/ieeefp.h
+++ b/libjava/classpath/native/fdlibm/ieeefp.h
@@ -4,6 +4,14 @@
 #ifndef __IEEE_BIG_ENDIAN
 #ifndef __IEEE_LITTLE_ENDIAN
 
+#ifdef __aarch64__
+#ifdef __AARCH64EB__
+#define __IEEE_BIG_ENDIAN
+#else
+#define __IEEE_LITTLE_ENDIAN
+#endif
+#endif
+
 #ifdef __alpha__
 #define __IEEE_LITTLE_ENDIAN
 #endif
diff --git a/libjava/configure.host b/libjava/configure.host
index 0c3b41c..f2d1bb5 100644
--- a/libjava/configure.host
+++ b/libjava/configure.host
@@ -81,6 +81,11 @@ ATOMICSPEC=
 
 # This case statement supports per-CPU defaults.
 case "${host}" in
+  aarch64*-linux*)
+	libgcj_interpreter=yes
+	sysdeps_dir=aarch64
+	ATOMICSPEC=-fuse-atomic-builtins
+	;;
   arm*-elf)
 	with_libffi_default=no
 	PROCESS=Ecos
@@ -289,6 +294,12 @@ EOF
 	sysdeps_dir=i386
 	DIVIDESPEC=-f%{m32:no-}use-divide-subroutine
 	;;
+  aarch64*-linux* )
+	slow_pthread_self=no
+	can_unwind_signal=no
+	CHECKREFSPEC=-fcheck-references
+	DIVIDESPEC=-fuse-divide-subroutine
+	;;
   arm*-linux* )
 	slow_pthread_self=no
 	can_unwind_signal=no
