diff -up gcc-13-20240323/libstdc++-v3/include/bits/stl_construct.h.omv~ gcc-13-20240323/libstdc++-v3/include/bits/stl_construct.h
--- gcc-13-20240323/libstdc++-v3/include/bits/stl_construct.h.omv~	2024-04-02 16:05:16.005754956 +0200
+++ gcc-13-20240323/libstdc++-v3/include/bits/stl_construct.h	2024-04-02 16:06:40.810791752 +0200
@@ -90,10 +90,9 @@ _GLIBCXX_BEGIN_NAMESPACE_VERSION
 
 #if __cplusplus >= 202002L
   template<typename _Tp, typename... _Args>
-    constexpr auto
+    constexpr _Tp*
     construct_at(_Tp* __location, _Args&&... __args)
     noexcept(noexcept(::new((void*)0) _Tp(std::declval<_Args>()...)))
-    -> decltype(::new((void*)0) _Tp(std::declval<_Args>()...))
     { return ::new((void*)__location) _Tp(std::forward<_Args>(__args)...); }
 #endif // C++20
 #endif// C++17
