diff -up gcc-14.1.0-RC-20240430/libstdc++-v3/include/bits/stl_construct.h.3~ gcc-14.1.0-RC-20240430/libstdc++-v3/include/bits/stl_construct.h
--- gcc-14.1.0-RC-20240430/libstdc++-v3/include/bits/stl_construct.h.3~	2024-04-30 21:06:17.387222211 +0200
+++ gcc-14.1.0-RC-20240430/libstdc++-v3/include/bits/stl_construct.h	2024-04-30 21:07:24.740011057 +0200
@@ -90,10 +90,9 @@ _GLIBCXX_BEGIN_NAMESPACE_VERSION
 
 #if __cpp_constexpr_dynamic_alloc // >= C++20
   template<typename _Tp, typename... _Args>
-    constexpr auto
+    constexpr _Tp*
     construct_at(_Tp* __location, _Args&&... __args)
     noexcept(noexcept(::new((void*)0) _Tp(std::declval<_Args>()...)))
-    -> decltype(::new((void*)0) _Tp(std::declval<_Args>()...))
     { return ::new((void*)__location) _Tp(std::forward<_Args>(__args)...); }
 #endif // C++20
 #endif// C++17
